src/mgmt/srv_pool.c

```diff
@@ -41,21 +40,9 @@ ds_mgmt_tgt_pool_destroy(uuid_t pool_uuid, crt_group_t *grp)
 	int				topo;
 	int				rc;
 
-	pool = ds_pool_lookup(pool_uuid);
-	if (pool != NULL) {
-		/* This may not be the pool leader node, so down targets
-		 * may not be updated, then the following collective RPC
-		 * might be timeout. XXX
-		 */
-		ABT_rwlock_rdlock(pool->sp_lock);
-		rc = map_ranks_init(pool->sp_map, MAP_RANKS_DOWN, &excluded);
-		ABT_rwlock_unlock(pool->sp_lock);
-		if (rc != 0) {
-			D_ERROR(DF_UUID": failed to create rank list: %d\n",
-				DP_UUID(pool->sp_uuid), rc);
-			return rc;
-		}
-	}
+	rc = ds_pool_get_ranks_by_uuid(pool_uuid, MAP_RANKS_DOWN, &excluded);
+	if (rc)
+		return rc;
```

Comment:
This is more of temporary hack, I guess? This and the pool create path suffer from the misunderstanding that the MS is part of every pool. That is not the case.

Reply:
Sorry, what do you mean MS here?  Hmm, this is just for function generation.  Though I am not sure exclude the ranks should be done by caller or callee during pool destroy.

Reply:
MS == Management service